bytebuffer/bytebufferpool.go

```diff
@@ -0,0 +1,20 @@
+package bytebuffer
+
+import "github.com/valyala/bytebufferpool"
+
+var (
+	// GetLen returns byte buffer with fixed length from the pool, exported from httprouter/bytebuffer.
+	GetLen = func(n int) *bytebufferpool.ByteBuffer {
+		b := bytebufferpool.Get()
+		if cap(b.B) < n {
+			b.B = make([]byte, n)
+			return b
+		}
+		b.B = b.B[:n]
```

Comment:
Is this resizing needed? What of the case `cap(b.B) == n` then creating a new slice for an unneeded resizing seems wasteful and it should also be a quite common case where fixed size slices are used.

Reply:
bytebufferpool.Get() returns an empty byte buffer from the pool, which means its length will be 0 so I have to reslice to make the length of byte slice equal to the given `n`.

Reply:
Ok but the case where they are equal? Then no resizing is needed right?

Reply:
OK I get your point, I will add one more if case when it is equal to `n` then it will not reslice.

Reply:
No, even if n==cap, it should be resliced cuz the byte slice's length is 0.

Reply:
Ok.